music21/romanText/writeRoman.py

```diff
@@ -0,0 +1,367 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+# Name:         romanText/writeRoman.py
+# Purpose:      Writes RomanText files given a music21 stream of Roman numerals
+#
+# Authors:      Mark Gotham
+#
+# Copyright:    Copyright © 2020 Michael Scott Cuthbert and the music21 Project
+# License:      BSD, see license.txt
+# ------------------------------------------------------------------------------
+'''
+Writer for the 'RomanText' format (Tymoczko, Gotham, Cuthbert, & Ariza ISMIR 2019)
+given a music21 stream (score or part) of Roman numerals as input.
+'''
+
+import fractions
+import unittest
+
+from typing import Optional, Union
+
+from music21 import metadata
+from music21 import roman
+from music21 import stream
+
+
+# ------------------------------------------------------------------------------
+
+class RnWriter:
+    '''
+    Extracts the relevant information from a stream of Roman numeral objects for
+    writing to text files in the 'RomanText' format.
+
```

Comment:
I am not very good at writing, but maybe something like this would be good in here:

---

`Part` and `Score` streams are supported. When the stream is a `Part`, its inner Roman numeral objects are used to generate the output `RomanText`. When the stream is a `Score`, the Roman numeral objects within the *first* `Part` of the `Score` are used to generate the output `RomanText`.

---

Reply:
Good call! More explanation and demonstrations coming right up for the class itself.